### PR TITLE
Team6api

### PR DESCRIPTION
**Pull request by tttr222**
I've added all the GET api endpoints for checkin history.

Only thing left is handling of POST request for: */inventory/:tag/history*

Also added a table seeder for checkin history.

### CODE REVIEW

File: database/seeds/DatabaseSeeder.php [hunk 1]
```diff
@@ -98,6 +100,30 @@ public function run()
...
+		$rooms = Room::all();
+		$items = Item::all();
+		srand(1);
+		foreach($rooms as $room) {
+			foreach($items as $item) {
+				for($i = 0; $i < rand(1,4); $i++) {
+					CheckIn::create([
+						'room_id'	=> $room->id,
+						'item_id'	=> $item->id,
+						'created_at' => rand(time()/2,time())
```
Comment (zachmay) [inline database/seeds/DatabaseSeeder.php]: We were looking at the DB table before we looked at the Seeder script... we had a chuckle when we realized where the dates came from. :smile:

File: app/Http/Controllers/Api/CheckInController.php [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+		// if inadequate input, set response code to 400 bad request
+		if($tag == null || trim($tag) == '' || !in_array($format,array('json','csv'))) 
+			return new Response(null,400);
+	
+		try {
+			// get the item with the specified asset tag
+			$item = Item::where('asset_tag',$tag)->firstOrFail();
+			
+			// get the checkin history for the item and set response
+			$checkins = CheckIn::where('item_id',$item->id)->get();
```
Comment (zachmay) [inline app/Http/Controllers/Api/CheckInController.php]: We should pull these results back, sorted based on the check-in record's created date. That wasn't in the spec, but it makes sense. Let's put the newest check-in first in the list.

File: app/Http/Controllers/Api/CheckInController.php [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+						$wroteHeader = true;
+					}
+					
+					$csvText .= implode(',',array_values($value)) . "\n";
+				}
+				
+				$response = new Response($csvText,200);
+				$response->header('Content-Type','text/csv');
+			}
+		} catch(ModelNotFoundException $e) {
```
Comment (zachmay) [inline app/Http/Controllers/Api/CheckInController.php]: What are we expecting to catch here? Can't we just use `findOrFail()` to have Laravel throw a 404 response if the item doesn't exist?

File: app/Http/Controllers/Api/CheckInController.php [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+					if(!$wroteHeader) {
+						$csvText .= implode(',',array_keys($value)) . "\n";
+						$wroteHeader = true;
+					}
+					
+					$csvText .= implode(',',array_values($value)) . "\n";
+				}
+				
+				$response = new Response($csvText,200);
+				$response->header('Content-Type','text/csv');
```
Comment (zachmay) [inline app/Http/Controllers/Api/CheckInController.php]: The general idea of REST is to use HTTP's content type headers (specifically `Accept`) to specify different output formats, but this is cool.

File: app/Http/Controllers/Api/CheckInController.php [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+	}
+
+}
+    /**
+     * POST /inventory/:tag/history
+     * Create a new check-in resource for the specified option, 
+	 based on the data in the request body (in particular, the room ID).
+     */
+function postHistory($tag = null, Request $request){
+	echo "posted mother fucker";
```
Comment (zachmay) [inline app/Http/Controllers/Api/CheckInController.php]: Heh heh.

**Comment by zachmay** [comment]
Good work guys! I've made some comments. Take a look at those, ask questions.

File: app/Http/Controllers/Api/CheckInController.php (hunk 3 above)
Comment (tttr222) [inline app/Http/Controllers/Api/CheckInController.php]: From what I understand, firstOrFail() throws ModelNotFoundException if the item is not found. So I am catching that and returning a 404 response. Can you elaborate on your statement? It sounds like there's little difference in the method you describe and whats in the code.

File: app/Http/Controllers/Api/CheckInController.php (hunk 2 above)
Comment (tttr222) [inline app/Http/Controllers/Api/CheckInController.php]: ok done

File: app/Http/Controllers/Api/CheckInController.php (hunk 3 above)
Comment (zachmay) [inline app/Http/Controllers/Api/CheckInController.php]: @tttr222 No, you've got it right. I thought `findOrFail()` triggered the 404 by itself. Sorry.

File: config/app.php [hunk 6]
```diff
@@ -135,6 +135,7 @@
 		'Illuminate\Translation\TranslationServiceProvider',
 		'Illuminate\Validation\ValidationServiceProvider',
 		'Illuminate\View\ViewServiceProvider',
+		'Illuminate\Html\HtmlServiceProvider',
```
Comment (zachmay) [inline config/app.php]: What is this for?

**Comment by zachmay** [comment]
OK, we figured out why I was getting the message about a class not being found. It's generally a good idea to add a note if you add new dependencies (like @akm-sabbir did in [composer.json](https://github.com/zachmay/inventory616/commit/fb094e03763c9a670c28f1c95c150f5ac17603c0)).

**Comment by akm-sabbir** [comment]
Zack i am sorry but i think there is a another branch team6api-working i have recently pushed updated code there.

**Comment by akm-sabbir** [comment]
hi Zack i made latest commit in team6api-working branch. you can look at
the codes there. just ignore the branch team6api now.
thanks and br
Sabbir

**Comment by zachmay** [comment]
@akm-sabbir Please make a pull request for that branch.

**Pull request closed without merging**